src/main/java/org/commcare/core/network/CommCareNetworkServiceGenerator.java

```diff
@@ -39,13 +49,40 @@
         return response;
     };
 
+    private static Interceptor driftInterceptor = new Interceptor() {
+        @Override
+        public Response intercept(Chain chain) throws IOException {
+            Request request = chain.request();
+            Response response = chain.proceed(request);
+            ICommCarePreferenceManager commCarePreferenceManager = CommCarePreferenceManagerFactory.getCommCarePreferenceManager();
+            if (commCarePreferenceManager != null) {
+                String serverDate = response.header("date");
+                try {
+                    long serverTimeInMillis = new SimpleDateFormat("EEE, dd MMM yyyy hh:mm:ss zzz").parse(serverDate).getTime();
+                    long now = new Date().getTime();
+                    long currentDrfit = (now - serverTimeInMillis) / HOUR_IN_MS;
+                    commCarePreferenceManager.putLong(CURRENT_DRIFT, currentDrfit);
+                    long maxDrfitSinceLastHeartbeat = commCarePreferenceManager.getLong(MAX_DRIFT_SINCE_LAST_HEARTBEAT, 0);
+                    currentDrfit *= currentDrfit < 0 ? -1 : 1; // make it positive to calculate max drift
```

Comment:
It looks like you should also be doing this before you store `CURRENT_DRIFT ` in app prefs, because in `HomeScreenBaseActivity::checkForDrift`, you're checking `if (getCurrentDrift() > 0)`

Reply:
(Also I think it just makes sense conceptually so think of drift as being an absolute value)

Reply:
Ahh I wanted to be able to tell whether device time is ahead or behind. I am going to change the  `getCurrentDrift() > 0` to `getCurrentDrift() != 0` instead ?

Reply:
That also works, but I don't see anywhere in the code where you're actually checking whether device time is ahead or behind. Do you mean that we want to be able to do that on HQ?

Reply:
Yup, I meant on HQ.